slik_wrangler/preprocessing.py

```diff
@@ -56,52 +56,68 @@ def bin_age(dataframe=None, age_col=None, add_prefix=True):
     return data
 
 
-def change_case(dataframe=None ,column=None,case='lower'):
-
+def change_case(dataframe, columns=None, case='lower', inplace=False):
     """
     Change the case of a pandas series to either upper or lower case
 
     Parameters
     ----------
     dataframe: Dataframe or named Series
-    
-    column: str.
-        The column to perform the operation on
-    
-    case: str. Default is set to lower 
+
+    columns: str, list
+        The column or list of columns to perform the operation on
+
+    case: str. Default is set to lower
         Indicates the type of operation to perform
 
+    inplace: bool. Default is set to False
+        Indicates if changes should by made within the dataframe or not.
+
     Returns
     -------
     Pandas Dataframe:
     """
-        
-    if dataframe is None:
-        raise ValueError("data: Expecting a DataFrame or Series, got 'None'")
-        
-    if not isinstance(column, str):
-        errstr = f'The given type for column is {type(column).__name__}. Expected type is a string'
-        raise TypeError(errstr)
 
-    df = dataframe.copy()
+    if type(dataframe) != pd.DataFrame and type(dataframe) != pd.Series:
+        raise ValueError(
+            "data: Expecting a DataFrame or Series, got None"
+        )
+
+    if isinstance(columns, str):
+        columns = [columns]
+
+    if not isinstance(columns, list):
+        raise TypeError(
+            f'The given type for column is {type(columns).__name__}. Expected type is a string or list'
+        )
+
+    if not inplace:
+        dataframe = dataframe.copy()
 
     def capitalize_case(x):
-        list_of_str = x.split(' ')
-        new_list =[]
-        for item in list_of_str:
-            new_list.append(item.capitalize())
-        JoinedStr = ' '.join(new_list)
-        return JoinedStr
-    
-    if case == 'lower':
-        df[column] = df[column].apply(lambda x: x.lower())
-        return df
-    elif case == 'upper':
-        df[column] = df[column].apply(lambda x: x.upper())
-        return df
-    elif case=='capitalize':
-        df[column] = df[column].apply(lambda x: capitalize_case(x))
-        return df
+        """
+        Capitalize Case Function
```

Comment:
Can we make this a bit more descriptive?

Reply:
Sure we can, would do that now!